src/priceHistory/priceHistoryChart.tsx

```diff
@@ -0,0 +1,321 @@
+        result.push(
+          <ChartAwareSvgText
+            x={x}
+            y={y}
+            key={idx}
+            value={chartData[lowestRateIdx].displayValue}
+            position={'bottom'}
+            chartWidth={chartWidth}
+          />
+        )
+        break
+    }
+
+    switch (result.length) {
+      case 0:
+        return null
+      case 1:
+        return result[0]
+      default:
+        return <>{result}</>
+    }
+  }
+}
+
+interface PriceHistoryChartProps {
+  tokenId: string
+  containerStyle?: ViewStyle
+  testID?: string
+  chartPadding?: number
+  color?: colors
+}
+
+export default function PriceHistoryChart({
+  tokenId,
+  containerStyle,
+  testID,
+  chartPadding,
+  color = colors.black,
+}: PriceHistoryChartProps) {
+  const priceHistoryPrices = useSelector((state: RootState) =>
+    priceHistoryPricesSelector(state, tokenId)
+  )
+
+  const priceHistoryStatus = useSelector((state: RootState) =>
+    priceHistoryStatusSelector(state, tokenId)
+  )
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  const localExchangeRate = useSelector(usdToLocalCurrencyRateSelector)
+  const dispatch = useDispatch()
+
+  const dollarsToLocal = useCallback(
+    (amount: BigNumber.Value | null) =>
+      convertDollarsToLocalAmount(amount, localCurrencyCode ? localExchangeRate : 1),
+    [localExchangeRate]
+  )
+  const displayLocalCurrency = useCallback(
+    (amount: BigNumber.Value) =>
+      getLocalCurrencyDisplayValue(amount, localCurrencyCode || LocalCurrencyCode.USD, true),
+    [localCurrencyCode]
+  )
+
+  useEffect(() => {
+    const latestTimestamp = priceHistoryPrices.at(-1)?.priceFetchedAt ?? 0
+    if (priceHistoryPrices.length > 0 && latestTimestamp > Date.now() - ONE_HOUR_IN_MILLIS) {
+      return
+    }
+
+    dispatch(
+      fetchPriceHistoryStart({
+        tokenId,
+        startTimestamp: Date.now() - ONE_DAY_IN_MILLIS * 30,
+        endTimestamp: Date.now(),
+      })
+    )
+  }, [])
+
+  if (priceHistoryStatus === 'loading' && priceHistoryPrices.length === 0) {
+    return <Loader />
+  } else if (priceHistoryPrices.length === 0) {
+    return null
+  }
+
+  const chartData = []
+  let lastTimestampAdded
+  for (let i = 0; i < priceHistoryPrices.length; i++) {
+    // Only grab one price per chart step & the most recent price
+    if (
+      lastTimestampAdded &&
+      priceHistoryPrices[i].priceFetchedAt - lastTimestampAdded <
+        ONE_HOUR_IN_MILLIS * CHART_STEP_IN_HOURS &&
+      i !== priceHistoryPrices.length - 1
+    ) {
+      continue
+    } else {
+      lastTimestampAdded = priceHistoryPrices[i].priceFetchedAt
+      const { priceUsd } = priceHistoryPrices[i]
+      const localAmount = dollarsToLocal(priceUsd)
+      chartData.push({
+        amount: localAmount ? localAmount.toNumber() : 0,
+        displayValue: localAmount ? displayLocalCurrency(localAmount) : '',
+        priceFetchedAt: priceHistoryPrices[i].priceFetchedAt,
+      })
+    }
+  }
```

Comment:
This is needed as the biggest slow down on page renders was loading all the data points into the chart and this cut down render times significantly. We could move this logic to blockchain-api if we wanted, but handling it on the frontend didn't appear to be a bottleneck.